src/angle_distribution.F90

```diff
@@ -0,0 +1,55 @@
+module angle_distribution
+
+  use constants, only: ZERO, ONE
+  use distribution_univariate, only: DistributionContainer
+  use random_lcg, only: prn
+  use search, only: binary_search
+
+  implicit none
+  private
+
+  type, public :: AngleDistribution
+    real(8), allocatable :: energy(:)
+    type(DistributionContainer), allocatable :: distribution(:)
+  contains
+    procedure :: sample => angle_sample
+  end type AngleDistribution
+
+contains
+
+  function angle_sample(this, E) result(mu)
```

Comment:
This is pretty self-explanatory, but put a comment block for it anyway.  Maybe something like "Sample a polar cosine from the given energy-dependent AngleDistribution"